android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt

```diff
@@ -90,15 +81,7 @@ open class ScreenContainer<T : ScreenFragment>(context: Context?) : ViewGroup(co
         get() = mParentScreenFragment != null
 
     protected fun markUpdated() {
```

Comment:
I'd suggest we get rid of this naming policy now that all updtes are executed immediately. There is no need to a method named "markUpdated" if it does not in fact mark anything and just executes the update. What I think would make the most sense is if we remove `markUpdated` and `updateIfNeeded` methods, then instead call `performUpdate` directly. In turn, `performUpdate` should do `mAttached` and other checks (that are currently in `updateIfNeeded` and all the logic from `performUpdate` should be moved to `onUpdate` that can be overridden by native stack implementation hence allow for different transactions to be instantiated etc.